torch/csrc/generic/StorageSharing.cpp

```diff
@@ -557,33 +574,31 @@ PyObject * THPStorage_(sharedFd)(PyObject *_self, PyObject *noargs)
 PyObject * THPStorage_(isShared)(PyObject *_self, PyObject *noargs)
 {
   auto self = (THPStorage*)_self;
-#ifdef THC_GENERIC_FILE
-  Py_RETURN_TRUE;
-#else
+  if (self->cdata->device_type() == at::kCUDA) {
+    Py_RETURN_TRUE;
+  }
   if (at::MapAllocator::fromDataPtr(self->cdata->data_ptr()) ||
       THManagedMapAllocator::fromDataPtr(self->cdata->data_ptr())) {
     Py_RETURN_TRUE;
   } else {
     Py_RETURN_FALSE;
   }
-#endif
 }
 
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,cppcoreguidelines-avoid-non-const-global-variables)
 static PyMethodDef THPStorage_(sharingMethods)[] = {
   {"_new_with_weak_ptr", THPStorage_(newWithWeakPtr), METH_O | METH_CLASS, nullptr},
-#ifdef THC_GENERIC_FILE
   {"_share_cuda_", THPStorage_(shareCuda), METH_NOARGS, nullptr},
   {"_new_shared_cuda", THPStorage_(newSharedCuda), METH_VARARGS | METH_STATIC, nullptr},
+  // TODO: Probably should rename all the CUDA/CPU-specific functions to append
```

Comment:
@ezyang, do you agree with appending `_cpu` or `_cuda` to these functions?

Reply:
I've added this as a separate commit, so I can easily revert it if we need to

Reply:
yeah sure, my concern is mostly around BC breakage, if that's not a problem go for it